src/goto-programs/goto_function.h

```diff
@@ -117,7 +117,59 @@ class goto_functiont
   /// reported via DATA_INVARIANT violations or exceptions.
   void validate(const namespacet &ns, const validation_modet vm) const
   {
-    body.validate(ns, vm);
+    auto type_symbol_finder =
```

Comment:
These seem very useful utilities - can we move them to/reconcile them with `util/find_symbols.{h,cpp}`?